src/silx/app/view/CustomPlotSelectionWindow.py

```diff
@@ -548,6 +583,33 @@ def addClearAction(self, treeView: qt.QTreeView):
         self.addAction(clearAction)
 
 
+class DropOverlay(qt.QWidget):
+    """Overlay widget for displaying drop zones on the plot."""
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        self.setAttribute(qt.Qt.WA_TransparentForMouseEvents)
+        self.setAttribute(qt.Qt.WA_NoSystemBackground)
+        self.hide()
+
+    def showOverlay(self, rect):
+        """Show the overlay at the given rectangle."""
+        self.setGeometry(rect)
+        self.show()
```

Comment:
Not really needed IMO, you could call `setGeometry` directly from `_showDropOverlay` and `show` from `dragEnterEvent`